spec/pg/encoder_spec.cr

```diff
@@ -1,18 +1,19 @@
 require "../spec_helper"
 
+enum EncoderSpec::Status
+  Open
+  Closed
+end
+
 private def test_insert_and_read(datatype, value, file = __FILE__, line = __LINE__)
   it "inserts #{datatype}", file, line do
     PG_DB.exec "drop table if exists test_table"
     PG_DB.exec "create table test_table (v #{datatype})"
+    PG_DB.exec "insert into test_table values ($1)", args: [value]
 
     # Read casting the value
-    PG_DB.exec "insert into test_table values ($1)", args: [value]
     actual_value = PG_DB.query_one "select v from test_table", as: value.class
     actual_value.should eq(value)
-
-    # Read without casting the value
-    actual_value = PG_DB.query_one "select v from test_table", &.read
-    actual_value.should eq(value)
```

Comment:
I removed this section because it assumes the value will be used in its raw form when retrieved. That assumption holds for data types that both Crystal and Postgres share but it breaks here.